### ISSUE TITLE
DEFAULT_FILE_NAME disturbs some Gradle task when use with alternative use case

### ISSUE THREAD

**Issue by YangJonghun** [numandev1/react-native-keys#26]
### Current behavior
Hi,
we use this library with below config
```
project.ext.keyFiles = [
    debug: "keys.dev.json",
    staging: "keys.stage.json",
    release: "keys.prod.json",
]
```
but above config makes gradle task(like `./gradlew clean`) failing
because we can't custom DEFAULT_FILE_NAME value(keys.development.json) within your lib and clean task doesn't recognize our gradle config value
(only solution is patching DEFAULT_FILE_NAME value)

### Expected behavior
Gradle tasks are work well

### Platform

- [X] Android
- [ ] iOS

### React Native Version
any version

**Comment by numandev1** [comment]
@YangJonghun `DEFAULT_FILE_NAME` is fallback, it triggers when user dont define any `keyFiles`, can you paste your `build.gradle` file code here?

**Comment by YangJonghun** [comment]
it is simple to reproduce
1. `npx react-native@latest init AwesomeProject`
2. add react-native-keys
3. add json files(without file named `keys.development.json`) in root path, I used below sample key files
  ```
  keys.dev.json
  keys.prod.json
  keys.staging.json
  ```
4. add below code within `android/app/build.gradle`
  ```
  project.ext.keyFiles = [
      debug: "keys.dev.json",
      staging: "keys.stage.json",
      release: "keys.prod.json",
  ]
  apply from: project(':react-native-keys').projectDir.getPath() + "/RNKeys.gradle"
  ```
5. run `cd android && ./gradlew clean`
6. and then you can see below error.
```
> Configure project :app
keys.development.json
**************************
*** Missing KEYs file ****
**************************
```

build and run works properly, but some Gradle task will fail.

**Comment by numandev1** [comment]
@YangJonghun thanks a lot for point out this issue

**Comment by numandev1** [comment]
@YangJonghun can you check this issue with `0.5.7` version?

**Comment by YangJonghun** [comment]
@numandev1 
Thanks for your works
however, this fix introduces another bug.
when we use key value within groovy.
it doesn't work.

you can test apply below diff and run `./gradlew clean`
```diff
project.ext.keyFiles = [
    debug: "keys.dev.json",
    staging: "keys.stage.json",
    release: "keys.prod.json",
]
apply from: project(':react-native-keys').projectDir.getPath() + "/RNKeys.gradle"

+ def value = project.keys.get("KEY")

android {
    ndkVersion rootProject.ext.ndkVersion
```

then you will encounter the following error
```
* Where:
Build file '/Users/yangjonghun/Desktop/AwesomeProject/android/app/build.gradle' line: 78

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not get unknown property 'keys' for project ':app' of type org.gradle.api.Project.
```

**Comment by YangJonghun** [comment]
@numandev1 
Thanks for your works
however, this fix introduces another bug.
when we use key value within groovy.
it doesn't work.

you can test with below diff and run `./gradlew clean`
```diff
project.ext.keyFiles = [
    debug: "keys.dev.json",
    staging: "keys.stage.json",
    release: "keys.prod.json",
]
apply from: project(':react-native-keys').projectDir.getPath() + "/RNKeys.gradle"

+ def value = project.keys.get("KEY")

android {
    ndkVersion rootProject.ext.ndkVersion
```

then you will encounter the following error
```
* Where:
Build file '/~~~~app path~~~/AwesomeProject/android/app/build.gradle' line: 78

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not get unknown property 'keys' for project ':app' of type org.gradle.api.Project.
```

**Comment by YangJonghun** [comment]
below patch works fine (because gradle clean task doesn't have any flavor)
```diff
     println(keysFile)
     def keys = [:]
     File f = new File("$project.rootDir/../$keysFile");
-    if (!f.exists()) {
-      println("**************************")
-      println("*** Missing KEYs file ****")
-      println("**************************")
-      return;
-    }
 
     if (f.exists()) {
       def parsedJson = new groovy.json.JsonSlurper().parseText(f.text)
-//     println(parsedJson.public)
       parsedJson.public.each { key, value ->
         keys.put(key, value.replace('"', '\\"'))
       }
     } else {
       println("**************************")
       println("*** Missing KEYs file ****")
       println("**************************")
+      return;
     }
 
     project.ext.set("keys", keys)

     android {
       defaultConfig {
         keys.each { k, v ->
           def escaped = v.replaceAll("%","\\\\u0025")
           buildConfigField "String", k, "\"$v\""
           resValue "string", k, "\"$escaped\""
         }
       }
     }
+  } else {
+    project.ext.set("keys", [:])
   }
 
 }

```

**Comment by numandev1** [comment]
@YangJonghun thanks for pointing out this issue, fixed in `0.5.8`, now you can use `0.5.8` version instead of patch
